### PR TITLE
Update send button accepts references

### PR DESCRIPTION
**Pull request by leonardoelias**

### CODE REVIEW

**Comment by cannikin** [comment]
Hi @leonardoelias thanks for this! 

Would you mind updating the docs for `<Submit>` along with a usage example? https://github.com/redwoodjs/redwood/tree/master/packages/web/src/form#submit

**Comment by cannikin** [comment]
Unless you think this is super common behavior and no explanation is necessary?

**Comment by lrosa007** [comment]
@cannikin it's pretty "normal"/"common" to allow simple components, that get wrapped like this, to receive a ref and forward it along to the single child. It's arguably good practice when exposing small utility components. Doesn't do much anyways expect for allow a ref to be passed down to the underlying button.

```js
const MySomething = () => {
  const refThingy = useRef();

  const onSubmit = () => {
    alert("much click");
  };
  
  useEffect(() => {
    setTimeout(() => {
      // without the forward ref this wouldn't be possible
     // no clue why someone would want to but could totally happen
      refThingy.current.click();
    }, 10000);
  }, [])

  return <Submit ref={ref} onSubmit={onSubmit}>Click Here</Submit>;
};
```

**Comment by cannikin** [comment]
Sounds good to me, thanks! v0.2.0 just went out but this'll make it into v0.2.1 or v0.3.0

**Comment by leonardoelias** [comment]
Thank you for the explanation @lrosa007 

Yes, this is very good, because we can concentrate when we open a modal, for example.
There are a few extras when working with libraries like Reakit / Emotion.

Am I pondering to put this on all components of the form? What do you think ?

React documentation, talk about it

```
However, it can be useful for some kinds of components, especially in reusable component libraries. The most common scenarios are described below.
```
https://reactjs.org/docs/forwarding-refs.html

**Comment by thedavidprice** [comment]
Hi @leonardoelias and @lrosa007 There is a new proposal about adding Directives. Was wondering if either of you have thoughts/reactions and could add to the discussion here:
https://community.redwoodjs.com/t/proposal-directives-helper-for-redwood/96

Thanks!

**Comment by lrosa007** [comment]
@thedavidprice

I’m not a fan of adding those kinds of custom things to the jsx but my opinion isn’t law or anything. .map for life 🙂❤️

**Comment by thedavidprice** [comment]
@lrosa007 that’s actually a very helpful perspective to add the the conversation. Again, no pressure if you don’t have time or feel comfortable engaging. Just know we are excited to create community, help people feel involved, and drive overall momentum through more than just our core team.

tl;dr:
community momentum == FTW

**Comment by lrosa007** [comment]
@thedavidprice I live on github, bother me as much as you would like. I’m even comfortable being directly asked to handle specific issues and tasks. I’ll gladly donate my time.

**Comment by thedavidprice** [comment]
@lrosa007 I look forward to taking you up on the offer! Any preferences for what you like to work on? We are trying to communicate priorities via pinned Issues. Current list:
- Storybook integration
- TS compatibility
- Unit Testing via Jest (for both CRA app and package repo)
